File: python/209_minimum_size_subarray_sum.py
class Solution:
    def minSubArrayLen(self, s: int, nums: List[int]) -> int:
        sum = 0
        left = 0
        ans = float('inf')

        for i, value in enumerate(nums):
            sum += value
            while sum >= s:
                ans = min(ans, i - left + 1)
                sum -= nums[left]
                left += 1
        return ans if ans != float('inf') else 0

File: 209_minimum_size_subarray_sum.java
class Solution {
    public int minSubArrayLen(int s, int[] nums) {
        if (nums == null || nums.length == 0) return 0;
        
        int ans = Integer.MAX_VALUE;
        int left = 0;
        int sum = 0;
        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];
            while (sum >= s) {
                ans = Math.min(ans, i - left + 1);
                sum -= nums[left++];
            }
        }
        return (ans != Integer.MAX_VALUE) ? ans : 0;
    }
}